netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -287,16 +305,14 @@ static ProtocolNegotiator createProtocolNegotiator(
 
     private boolean closed;
 
-    private NettyTransportFactory(Class<? extends Channel> channelType,
-                                  NegotiationType negotiationType,
-                                  ProtocolNegotiator protocolNegotiator,
-                                  SslContext sslContext,
-                                  EventLoopGroup group,
-                                  int flowControlWindow,
-                                  int maxMessageSize,
-                                  int maxHeaderListSize) {
+    private NettyTransportFactory(
+        Class<? extends Channel> channelType, Map<ChannelOption<?>, ?> channelOptions,
+        NegotiationType negotiationType, ProtocolNegotiator protocolNegotiator,
+        SslContext sslContext, EventLoopGroup group, int flowControlWindow, int maxMessageSize,
+        int maxHeaderListSize) {
       this.channelType = channelType;
       this.negotiationType = negotiationType;
+      this.channelOptions = new HashMap<ChannelOption<?>, Object>(channelOptions);
```

Comment:
I want the unmodifiableMap, because it _guarantees_ that we don't mutate it. The `?` helps prevent mutations, but isn't guaranteed. Casting and methods like `clear()` and `remove()` avoid the `?` restriction. Many people don't think twice when casting `?` to something else.